core/domain/exp_jobs_one_off.py

```diff
@@ -56,6 +57,30 @@ def map(exploration_model):
     def reduce(exp_id, list_of_exps):
         pass
 
+class ExplorationFirstPublishedOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job that finds first published datetime for all explorations."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationRightsSnapshotContentModel]
+
+    @staticmethod
+    def map(item):
+        if item.content['status'] == rights_manager.ACTIVITY_STATUS_PUBLIC:
+            snapshot_id = item.id
+            yield(
```

Comment:
As mentioned previously: add single space before '('; use single quotes in line below